preview_generator/manager.py

```diff
@@ -322,8 +322,13 @@ def _get_file_hash(self, file_path: str, size: ImgDims = None, page: int = None)
     def get_supported_mimetypes(self) -> typing.List[str]:
         return self._factory.get_supported_mimetypes()
 
-    def get_file_extension(self, mime: str) -> str:
+    def get_file_extension(self, mime: str) -> typing.Optional[str]:
```

Comment:
Idea here is to allow returning None in case file as no extension ?

Reply:
No, it's just no to lie about the return type of guess_extension which can return None:
```
extensions = self.guess_all_extensions(type, strict)
if not extensions:
    return None
return extensions[0]
```